static/flow/views/pi-selector-panel.js

```diff
@@ -741,6 +747,46 @@ var PiSelectorPanel = function(options) {
         clearTimeout(loadPiListTimer);
     }
 
+    //
+    // Start timer to determine if pi communication is taking too long
+    //
+    this.restartPiResponseTimer = function(runProgram, resetDeviceControls, refreshCallback) {
+        clearTimeout(piResponseTimer);
+        piResponseTimer = setTimeout(stopWaitingForResponse, piResponseTimerInterval, runProgram, resetDeviceControls, refreshCallback);
```

Comment:
Somehow I never knew that you could pass additional parameters to the function called by `setTimeout`.  Cool.

Reply:
Yeah, it's a nice feature.